artemis-core-client/src/main/java/org/apache/activemq/artemis/core/message/impl/CoreMessage.java

```diff
@@ -1271,4 +1271,22 @@ public Object getOwner() {
    public void setOwner(Object object) {
       this.owner = object;
    }
+
+   @Override
+   public String getStringBody() {
+      String body = null;
+
+      if (type == TEXT_TYPE) {
+         try {
+            SimpleString simpleBody = getBodyBuffer().readNullableSimpleString();
```

Comment:
You should return the position of the BodyBuffer after read. or duplicate the bodyBuffer here.

```
getBodyBuffer().duplicate().readNullableSimpleString();
```


otherwise the conversion will be broken if you send A Core Message, treat it with Path and receive with AMQP.

Reply:
Or even better.. use getReadOnlyBodyBuffer() instead of getbodyBuffer() here.

Reply:
so the code should be:

```java
SimpleString simpleBody = getDataBuffer().readNullableSimpleString();
```

Reply:
Large message are not supported. This is already noted in the documentation.

I'll update the buffer code with your suggestion.